dvc/scm/git/__init__.py

```diff
@@ -132,7 +133,25 @@ def ignore(self, path, in_curr_dir=False):
 
     @staticmethod
     def _add_entry_to_gitignore(entry, gitignore, ignore_list):
-        content = entry
+        def escape_gitignore_special_characters(gitignore_entry):
+            def exclamation_or_asterisk(matchobj):
+                if matchobj.group(0) == "!":
+                    return "\!"
+                elif matchobj.group(0) == "*":
+                    return "\*"
+                else:
+                    return matchobj.group(0)
+                gitignore_entry = re.sub(
+                    "^([!*])", exclamation_or_asterisk, gitignore_entry
+                )
+                escaped = gitignore_entry.translate(
+                    str.maketrans({"[": r"\[", "]": r"\]", "?": r"\?"})
+                )
+
+            return gitignore_entry
+
+        content = escape_gitignore_special_characters(entry)
```

Comment:
Looks like this function is a noop. `exclamation_or_asterisk()` is not even called. Looks like indentation issue: code starting with `gitignore_entry = re.sub(` should be unindented.

We are also only quoting `*` at the start of the string, which is probably not intended. I suggest using single regex substitute:
```python
content = re.sub(r'(^!|[*?\[\]])', r'\\\1', entry)
```

Reply:
Alright. Thanks for the suggestion --  Will look into this tomorrow morning